src/tclint/cli/tclsp.py

```diff
@@ -330,6 +331,57 @@ def format_range(ls: TclspServer, params: lsp.DocumentRangeFormattingParams):
     ]
 
 
+@server.feature(lsp.TEXT_DOCUMENT_DEFINITION)
+def goto_definition(ls: TclspServer, params: lsp.DefinitionParams):
```

Comment:
I thought of some edge cases for this feature that I might want addressed before it comes user-visible. I'm happy to do a slimmed-down PR with just the symbol table before solving these, in which case let's remove this for now (or perhaps feature flag it somehow?)